flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -342,6 +346,14 @@ private ContainerLaunchContext createTaskExecutorLaunchContext(Resource resource
 		FiniteDuration teRegistrationTimeout = new FiniteDuration(timeout, TimeUnit.SECONDS);
 		final Configuration taskManagerConfig = BootstrapTools.generateTaskManagerConfiguration(
 				flinkConfig, "", 0, 1, teRegistrationTimeout);
+		// Add resource profile of slots to task executor config. 
+		// For yarn, all slots in a task executor have same resource profile 
+		ByteArrayOutputStream output = new ByteArrayOutputStream();
+		ObjectOutputStream rpOutput = new ObjectOutputStream(output);
+		rpOutput.writeObject(getResourceProfile(priority.getPriority()));
+		rpOutput.close();
+		taskManagerConfig.setString(ConfigConstants.TASK_MANAGER_RESOURCE_PROFILE_KEY,
+				new String(Base64.encodeBase64(output.toByteArray())));
```

Comment:
I have to admit that I don't know the answer here. The documentation doesn't mention anything regarding preferences over the different options: https://hadoop.apache.org/docs/stable/hadoop-yarn/hadoop-yarn-site/WritingYarnApplications.html

I think we should use the 
- `flink-conf.yaml` for setting configuration parameters in Flink, but not for "internal data". Some users read the flink-conf.yaml files on the container for debugging purposes.
- Environment variables for small bits such as the application id (this is beneficial for getting the application name in the log4j2 logging configuration when using centralized logging)
- a separate file for larger serialized objects.

Ideally we don't need the third option.